src/edu/washington/escience/myria/expression/evaluate/PythonUDFEvaluator.java

```diff
@@ -327,15 +392,16 @@ private void writeToStream(
           LOGGER.info("date time not yet supported for python function ");
           break;
         case BLOB_TYPE:
-          dOut.writeInt(MyriaConstants.PythonType.BYTES.getVal());
-          // LOGGER.info("writing Bytebuffer to py process");
+          dOut.writeInt(MyriaConstants.PythonType.BLOB.getVal());
+
           ByteBuffer input = tb.getBlob(columnIdx, row);
+
           if (input != null && input.hasArray()) {
-            // LOGGER.info("input array buffer length" + input.array().length);
+
             dOut.writeInt(input.array().length);
             dOut.write(input.array());
           } else {
-            // LOGGER.info("input arraybuffer length is null");
+
             dOut.writeInt(MyriaConstants.NULL_LENGTH);
```

Comment:
I need to specifically pass null to tell python that the next parameter is null. Writing nothing breaks the protocol.